Library/Homebrew/test/test_utils.rb

```diff
@@ -223,4 +223,17 @@ def test_truncate_text_to_approximate_size
     s = truncate_text_to_approximate_size(long_s, n, :front_weight => 1.0)
     assert_equal(("x" * (n - glue.length)) + glue, s)
   end
+
+  def test_odeprecated
+    ARGV.stubs(:homebrew_developer?).returns false
+    begin
+      odeprecated("method", "replacement", :caller => ["#{HOMEBREW_LIBRARY}/Taps/homebrew/homebrew-core/"], :die => true)
+      assert false, "odeprecated did not throw an exception!"
+    rescue FormulaMethodDeprecatedError => e
+      assert_match "method", e.message
+      assert_match "replacement", e.message
+      assert_match "homebrew/homebrew-core", e.message
+      assert_match "homebrew/core", e.message
+    end
```

Comment:
You'll want to use `assert_raises` (that also returns the expected caught exception for further inspection) instead of this construct.